app/models/listing.rb

```diff
@@ -406,4 +409,10 @@ def answer_for(custom_field)
   def payment_required_at?(community)
     transaction_type.price_field? && community.payments_in_use?
   end
+
+  def currency=(currency)
+    raise ArgumentError.new "Invalid currency. Tried to change #{self[:currency]} to #{currency}." if self[:currency] && self[:currency] != currency
```

Comment:
What does this do? Did I understand right, it throws when ever someone is trying to change the currency?